scripts/install-multi-user.sh

```diff
@@ -576,18 +576,37 @@ create_directories() {
         # since this bit is cross-platform:
         # - first try with `command -vp` to try and find
         #   chown in the usual places
+        #   * to work around some sort of deficiency in
+        #     `command -p` in macOS bash 3.2, we also add
+        #     PATH="$(getconf PATH 2>/dev/null)". As long as
+        #     getconf is found, this should set a sane PATH
+        #     which `command -p` in bash 3.2 appears to use.
+        #     A bash with a properly-working `command -p`
+        #     should ignore this hard-set PATH in favor of
+        #     whatever it obtains internally. See
+        #     github.com/NixOS/nix/issues/5768#issuecomment-1018060400
         # - fall back on `command -v` which would find
         #   any chown on path
         # if we don't find one, the command is already
         # hiding behind || true, and the general state
         # should be one the user can repair once they
         # figure out where chown is...
-        local get_chr_own="$(command -vp chown)"
+        local get_chr_own="$(PATH="$(getconf PATH 2>/dev/null)" command -vp chown)"
```

Comment:
it seems like this should be the fallback if both `command -vp chown` and `command -v chown` fail to produce a valid path to `chown`. Otherwise you are overwriting a healthy `PATH` (and functioning `command`) when not necessary.

Reply:
That's roughly the intent... the `*` part of the comment earlier tries to explain, but maybe there are language tweaks that would clarify? The logic is roughly like:

- `command -vp` is supposed to check in the "default path"
    - bash 4.4+ do check the "default path", so they'll ignore the hard-set PATH here
    - bash 3.2 (in reality, probably earlier and up through 4.3) do not, so they end up using the hard-set PATH from getconf here.
        - if `getconf PATH` fails, it'll check an empty PATH. Not ideal, but it won't blow up.
        - if `getconf PATH` succeeds, it should print an ~equivalent "default path"
- if we didn't already find a chown, `command -v` will then fall back to the first `chown` on the existing PATH regardless of bash version